### PR TITLE
Fix AMI builds

### PR DESCRIPTION
**Pull request by saschagrunert**
An approach to make the AMI tests work again as well as minor cleanups and updates.

/test ami

### CODE REVIEW

File: contrib/test/integration/main.yml [hunk 1]
```diff
@@ -51,7 +51,7 @@
     - name: install Golang tools
       include: golang.yml
       vars:
-          version: "1.12.2"
+        version: "1.12.2"
```
Comment (TomSweeneyRedHat) [inline contrib/test/integration/main.yml]: 1.12.2 here and 1.12.6 above?  is that intentional?

Comment (saschagrunert) [reply contrib/test/integration/main.yml]: Hm, no I will fix it as soon as the build (hopefully) succeeds. Thanks :pray:

**Comment by saschagrunert** [comment]
It seems to build the AMI image but the e2e tests failed. So I hope we should be ready to go in a couple of hours. :innocent:

**Comment by saschagrunert** [comment]
gg/test integration_fedora

**Comment by saschagrunert** [comment]
If the last test succeeds I'd say hooray and kindly ask if we can get this in :innocent: 

@haircommander @mrunalp PTAL

**Comment by saschagrunert** [comment]
> The tests seem to like the change, but the commit message doesn't tell what the problem was and how it's intended to be fixed. Occasionally, I continue my crusade for informative git histories 

Yes, I will add a better commit message.

> On a more serious note: disabling SELinux is usually a symptomatic fix. What is the reason for disabling it?

I'm unsure and will give it another try with selinux. There were some flakes with the integration test suite but I'm not sure if the selinux disabling actually fixed anything. :man_shrugging: 

/test ami

**Comment by saschagrunert** [comment]
@vrothberg I think we have to disable selinux for the integration and cri-tests, I see this error:
```
time="2019-07-05 10:20:40.253153680Z" level=error msg="Container creation error: container_linux.go:345: starting container process caused \"process_linux.go:430: container init caused \\\"write /proc/self/attr/keycreate: permission denied\\\"\"\n"
```

**Comment by saschagrunert** [comment]
🥳 it finally builds

**Comment by rhatdan** [comment]
SELinux is still disabled?

**Comment by saschagrunert** [comment]
> SELinux is still disabled?

Yeah, let's see if the runc upgrade causes the issue.

**Comment by saschagrunert** [comment]
It succeeds again 🥳 the selinux issue is related to runc, and I will address this in a later PR.

@mrunalp @rhatdan @haircommander PTAL

**Comment by rhatdan** [comment]
LGTM
/test ami_rhel

**Comment by saschagrunert** [comment]
Finally, everything built. :)

**Comment by saschagrunert** [comment]
Thanks, do you know at which time the AMI images are available for all further tests?

**Comment by saschagrunert** [comment]
@mrunalp do you know at which time the AMI images are available for all further tests?

**Comment by giuseppe** [comment]
@saschagrunert it seems they are still not in place.  I've tried to rebase https://github.com/cri-o/cri-o/pull/2595 and it is still failing:/

**Comment by saschagrunert** [comment]
> @saschagrunert it seems they are still not in place. I've tried to rebase #2595 and it is still failing:/

I think the job definitions we need are available here: https://ci.openshift.redhat.com/jenkins/view/AMIs/
Since they seem to be running now we could wait until they finished and give it another try.

**Pull request merged**